app/controllers/hosts_controller.rb

```diff
@@ -454,6 +454,22 @@ def update_multiple_power_state
     redirect_back_or_to hosts_path
   end
 
+  def show_multiple_power_status
+    res = {}
+    if @hosts.respond_to? :map
+      res = @hosts.map do |host|
+        begin
+          { host.id => host.supports_power_and_running? }
+        rescue Foreman::Exception
+          {}
+        end
+      end.reduce(:merge)
+    else
+      logger.warn "Malformed host_ids parameter - not and array"
```

Comment:
Will we ever get here? if i'm not mistaken a malformed request will cause `find_multiple` to handle errors before this function is executed.